dashboards-observability/public/components/visualizations/charts/stats/stats.tsx

```diff
@@ -0,0 +1,441 @@
+      ? DefaultOrientation
+      : 'h';
+  const selectedTextMode = dataConfig?.chartStyles?.textMode || vis.textmode;
+  const textMode =
+    selectedTextMode === DefaultTextMode || selectedTextMode === 'values+names'
+      ? DefaultTextMode
+      : selectedTextMode;
+  const precisionValue = dataConfig?.chartStyles?.precisionValue || vis.precisionvalue;
+  const metricUnits =
+    dataConfig?.chartStyles?.metricUnits?.substring(0, STATS_METRIC_UNIT_SUBSTRING_LENGTH) || '';
+  const metricUnitsSize = valueSize - valueSize * STATS_REDUCE_METRIC_UNIT_SIZE_PERCENTAGE;
+  const isDarkMode = uiSettingsService.get('theme:darkMode');
+
+  const getRoundOf = (value: number, places: number) => {
+    return (Math.round(value * 10 ** precisionValue) / 10 ** precisionValue).toFixed(places);
+  };
+
+  const ZERO_ERROR_ANNOTATION = 0.01;
+  let autoChartLayout: any = {
+    xaxis: {
+      visible: false,
+      showgrid: false,
+      anchor: 'y1',
+    },
+    yaxis: {
+      visible: false,
+      showgrid: false,
+      anchor: 'x1',
+    },
+    annotations: [],
+  };
+
+  const selectedDimensionsData = dimensions
+    .map((dimension: any) => data[dimension.name])
+    .reduce((prev, cur) =>
+      prev.map((item: string | number, index: number) => `${item},<br>${cur[index]}`)
+    );
+
+  const createValueText = (value: string | number) =>
+    `<b>${value}${
+      metricUnits ? `<span style="font-size: ${metricUnitsSize}px"}> ${metricUnits}</span>` : ''
+    }</b>`;
+
+  const createAnnotationsAutoChartHorizontal = ({
+    label,
+    value,
+    index,
+    valueColor,
+  }: createAnnotationType) =>
+    textMode === 'values+names' || textMode === DefaultTextMode
+      ? [
+          {
+            ...STATS_ANNOTATION,
+            x: 0 + ZERO_ERROR_ANNOTATION,
+            y:
+              index > 0
+                ? (index + 1) / metricsLength - ZERO_ERROR_ANNOTATION
+                : 1 / metricsLength - ZERO_ERROR_ANNOTATION,
+            xanchor: 'left',
+            yanchor: 'bottom',
+            text: label,
+            font: {
+              size: titleSize,
+              color: isDarkMode ? COLOR_WHITE : COLOR_BLACK,
+              family: 'Roboto',
+            },
+            type: 'name',
+            metricValue: value,
+          },
+          {
+            ...STATS_ANNOTATION,
+            x: 1,
+            y:
+              index > 0
+                ? (index + 1) / metricsLength - ZERO_ERROR_ANNOTATION
+                : 1 / metricsLength - ZERO_ERROR_ANNOTATION,
+            xanchor: 'right',
+            yanchor: 'bottom',
+            text: createValueText(value),
+            font: {
+              size: valueSize,
+              color: valueColor,
+              family: 'Roboto',
+            },
+            type: 'value',
+            metricValue: value,
+          },
+        ]
+      : [
+          {
+            ...STATS_ANNOTATION,
+            x: 0.5,
+            y:
+              metricsLength === 1
```

Comment:
move this block separate with if else and avoid multiple ternary to improve readability.